gcloud-java-contrib/gcloud-java-nio/src/main/java/com/google/gcloud/storage/contrib/nio/CloudStorageObjectAttributes.java

```diff
@@ -131,25 +145,27 @@ public FileTime lastAccessTime() {
 
   @Override
   public Object fileKey() {
-    return metadata.getFilename();
+    return name;
```

Comment:
That should be fine then: we export one filesystem per bucket and fileKey only needs to be unique per-filesystem if I understand correctly.